### ISSUE TITLE
Suppress Reverse Navigation properties

### ISSUE THREAD

**Issue by jsmarsch** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#80]
I have a case where on just 1 table, I need to suppress reverse navigation properties from that table to all tables that have a foreign key to it.  Without going into details, it has to do with some legacy issues.

I found a recent question on the Questions/Answer page on Codeplex where this had been covered, however the suggested solution produces code that does not compile.

The suggestion was to do this:

```
Action<Table> WritePocoBaseClassBody = t =>
{
    t.ReverseNavigationProperty.RemoveAll(x => true);
    t.ReverseNavigationCtor.RemoveAll(x => true);
};
```

(I modified it with a check on the table name)

However, I found that when I do that, there is code generated in the // foreign keys section of the Fluent Configuration classes that attempts to reference the properties that have been suppressed.  This causes a compile error.

This will occur in the configuration classes of every class that owns the forward navigation (so there is no one place for me to go to fix it)

Since this occurs for every table that references the one that I want to suppress, the only workaround I see is to use the ForeignKeyFilter to eliminate both forward and backward navigation.

Here's an example of the code that is being created inside of the configuration objects of the referencing tables:

```
// Foreign keys
 HasOptional(a => a.BcegC).WithMany(b => b.BcegC).HasForeignKey(c => c.BcegCId).WillCascadeOnDelete(false); // fk_location_bceg

```

b.BcegC in the line above is the error, because it is referencing a reverse-nav property that has been suppressed.

**Comment by sjh37** [comment]
ForeignKeyFilter is the right option, but can be a pain if you have many references to this table you want to exclude all FK's for.

Try editing the file `EF.Reverse.POCO.Core.ttinclude` ProcessForeignKeys function.
near the top after it finds the pk/fk table. Check for the BcegC name there and exit.

**Comment by sjh37** [comment]
The ultimate answer would be to add a an array setting, containing a list of tables to exclude from FK's.

**Comment by jsmarsch** [comment]
Thank you -- I'm off on something else right now, but I will give it a shot when I'm back on that task.

**Comment by rory-a-hunt** [comment]
This page was really useful for me to get to the configuration I needed - which was a one way FK. E.g. no reverse navigation. It seems like EF supports this with an empty .WithMany() so I am tweaking the code gen to emit this. What would be the best way to flag an FK to suppress reverse navigation? A new property on the FK object?

Happy to commit back to the project once I get it sorted -  I think it's great!

-- required result
`// FK Column IS included`
`public string BcegC {get; set; }`
`// Foreign keys // Added with no reverse nav`
` HasOptional(a => a.BcegC).WithMany().HasForeignKey(c => c.BcegCId).WillCascadeOnDelete(false); // fk_location_bceg
`

**Comment by sjh37** [comment]
Hi @rory-a-hunt It has already been added to the code. Pull the latest code from GitHub and take a look at `IncludeReverseNavigation`
See #149

**Comment by rory-a-hunt** [comment]
Excellent - Just what I needed. I'll pull from this rather than VS and see about my lot pushing a donation to the project. Talk about just in time!
UPDATE : Hi Simon ( @sjh37 ) - when downloading the latest code, deleting Database.cs and regenerating I get Compiling transformation errors. Are there any more dependencies on this latest code?

**Comment by sjh37** [comment]
Released in v2.25
